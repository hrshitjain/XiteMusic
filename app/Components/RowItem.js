import React  from 'react';
import styles from '../Styles/HomeStyles';
import {Image, View, Text} from 'react-native'

class RowItem extends React.Component
{
    render(){
        return(
            <View style={styles.rowItem}>
                <View style={styles.detail}>
                    <Image style={styles.image} source={{uri:this.props.image}} />
                    <Text style={styles.name}>{this.props.name}</Text>
                    <Text style={styles.desc}>{this.props.artist}</Text>
                </View>
            </View>
        )
    }
}

export default RowItem;