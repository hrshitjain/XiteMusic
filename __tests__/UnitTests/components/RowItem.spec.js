import React from 'react';
import { render } from '@testing-library/react-native';
import RowItem from '../../../app/Components/RowItem';

const defaultProps = {
    image: "https://picsum.photos/id/237/200/300",
    name: "SRK",
    artist:'Sonu'
}

describe('RowItem component render', () =>{

    it('Snapshot test', () =>{
        const {toJSON} = render(<RowItem {...defaultProps}/>);
        expect(toJSON()).toMatchSnapshot();
    });
    it('Root View has 1 child', () => {
        const {toJSON} = render(<RowItem {...defaultProps}/>);
        expect(toJSON().children.length).toBe(1);
    });
    it('Component has 3 children', () => {
        const {toJSON} = render(<RowItem {...defaultProps}/>);
        expect(toJSON().children[0].children.length).toBe(3);
    });
    it('Artist Label is rendered',()=>{
        const {getAllByText} = render(<RowItem {...defaultProps}/>);
        expect(getAllByText(defaultProps.artist).length).toBe(1);
    });
    it('Music Label is rendered',()=>{
        const {getAllByText} = render(<RowItem {...defaultProps}/>);
        expect(getAllByText(defaultProps.name).length).toBe(1);
    });
});