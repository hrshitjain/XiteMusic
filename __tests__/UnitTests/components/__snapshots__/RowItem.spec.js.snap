// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowItem component render Snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColorL": "red",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://picsum.photos/id/237/200/300",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      SRK
    </Text>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Sonu
    </Text>
  </View>
</View>
`;
